File: app/src/main/java/com/androidacademy/msk/exerciseproject/di/module/AppModule.java
package com.androidacademy.msk.exerciseproject.di.module;

import android.content.Context;
import android.support.annotation.NonNull;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class AppModule {

    private Context appContext;

    public AppModule(@NonNull Context appContext) {
        this.appContext = appContext;
    }

    @Provides
    @Singleton
    @NonNull
    public Context provideContext() {
        return appContext;
    }
}

File: app/src/main/java/com/androidacademy/msk/exerciseproject/di/module/AppModule.kt
package com.androidacademy.msk.exerciseproject.di.module

import android.content.Context
import com.androidacademy.msk.exerciseproject.data.database.dao.NewsDao
import com.androidacademy.msk.exerciseproject.data.network.api.NYTimesApi
import toothpick.config.Module


class AppModule(context: Context) : Module() {

    init {
        bind(Context::class.java).toInstance(context)
        bind(NewsDao::class.java).toProvider(DbProvider::class.java)
        bind(NYTimesApi::class.java).toProvider(NetworkProvider::class.java)
    }
}